src/main/scala/firrtl/passes/clocklist/ClockListTransform.scala

```diff
@@ -51,9 +51,11 @@ Usage:
   }
 }
 
-class ClockListTransform extends Transform with RegisteredTransform {
-  def inputForm = LowForm
-  def outputForm = LowForm
+class ClockListTransform extends Transform with DependencyAPIMigration with RegisteredTransform {
+
+   override lazy val prerequisites = Forms.LowForm
+   override lazy val optionalPrerequisites = Seq.empty
+   override lazy val dependents = Forms.LowEmitters
```

Comment:
Is there a reason why sometimes you use `val` and sometimes `lazy val`? I actually think in all of these cases `def` would be _marginally_ better since the cost per call is very low. However, since we make so few transform objects I think the overhead of the `val` in the object is irrelevant anyway.

Reply:
I think `val` has the advantage of being consistent with the restriction that it be a constant value without relying on implicit conventions surrounding idempotent `def`s that have never really been strictly adhered to. But since this isn't enforced at the API level, the benefit of that consistency is negligible.

Reply:
Inconsistency on my part. I probably intended to always use `lazy val`.

Note: these fields are all shadowed by `private[options] lazy val`s. The `DependencyManager` is supposed to be using these to avoid a bunch of repeated conversions to sets. I looked and I'm actually not doing this.

So, what do you think? Use `def` everywhere? I'm fine with that.

Reply:
and don't forget the issues we've had with sharing data structures across multiple threads.

Reply:
I *think* that's only an issue on `object`s right? Should be fine since this is a `class` right?

Reply:
My guess is any mutable field in a class (def or lazy) is an issue if instances of that class are shared between threads.

Reply:
Effectively, the thread that issues the first reference will end up defining the value. This may (or may not) be what you want.

Reply:
I think there are two separate issues we're conflating here:

- the issue with ordering initialization in multi-threaded contexts (#1462, causes _hangs_ but not inconsistencies). This is a [dead,live]lock issue with _all_ members, not just collections / data structures.

- thread-unsafe modifications to mutable data structures (see #1057). This is a data race that causes inconsistencies and is only an issue with mutable data structures. This is also simple to fix with Java threading modifiers.

Reply:
Solution here: switching to `def` globally.